### PR TITLE
Add type Argument with some common parse function

### LINKED ISSUE #704: Add type Argument with some common parse function
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>
> /kind bug

 /kind feature

**What happened**:
PluginBuilder has function definition with code  Introduced in  #637 
> type PluginBuilder func(map[string]string) Plugin

Every plugin use it's own method to parse the arguments.

**What you expected to happen**:
a common support for arguments parse

**How to reproduce it (as minimally and precisely as possible)**:
I wanna to use a special type to replace it and add some common argument parse function.
> type Arguments map[string]string
> type PluginBuilder func(Arguments) Plugin

add function
> func (a Arguments) GetInt(ptr *int, key string)

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): N/A
- Cloud provider or hardware configuration:N/A
- OS (e.g. from /etc/os-release):N/A
- Kernel (e.g. `uname -a`):N/A
- Install tools:N/A
- Others:N/A

### PR DESCRIPTION
**Pull request by lmzqwer2**
**What this PR does / why we need it**:
1. add special type for plugin arguments
2. add common parse function GetInt

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #704

**Special notes for your reviewer**:

**Release note**:

```release-note

```

### CODE REVIEW

File: pkg/scheduler/framework/arguments.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+import (
+	"strconv"
+
+	"github.com/golang/glog"
+)
+
+type Arguments map[string]string
+
+func (a Arguments) GetInt(ptr *int, key string) {
```
Comment (k82cn) [inline pkg/scheduler/framework/arguments.go:27]: return value instead of pointer.

File: pkg/scheduler/framework/arguments.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+	argv, ok := a[key]
+	if !ok || argv == "" {
+		return
+	}
+
+	value, err := strconv.Atoi(argv)
+	if err != nil {
+		glog.Warningf("Could not parse argument: %s for key %s, with err %v", argv, key, err)
+		return
```
Comment (k82cn) [inline pkg/scheduler/framework/arguments.go:40]: ~~the return value is `(int, error)`~~

It's ok to me :)

File: pkg/scheduler/framework/arguments.go [hunk 3]
```diff
@@ -0,0 +1,28 @@
+package framework
```
Comment (k82cn) [inline pkg/scheduler/framework/arguments.go:17]: add license there.

File: pkg/scheduler/framework/arguments.go (hunk 1 above)
Comment (lmzqwer2) [reply pkg/scheduler/framework/arguments.go:27]: so, I think the definition should be.
```
var ERRNOTFOUND=errors.New("ARGUMENT NOT FOUND")

func (a Arguments) GetInt(key string) (value int, err error) { ... }
```
when the value of this key is not found, the return value should be 0, ERRNOTFOUND
when the value is in invalid format of int, the return value should be 0, err of parsing

File: pkg/scheduler/framework/arguments.go (hunk 3 above)
Comment (lmzqwer2) [reply pkg/scheduler/framework/arguments.go:17]: copy license from other file.

And I think the license may be outdated now? 
```
Copyright 2018 The Kubernetes Authors
```

File: pkg/scheduler/framework/arguments.go (hunk 2 above)
Comment (lmzqwer2) [reply pkg/scheduler/framework/arguments.go:40]: ok. :)

File: pkg/scheduler/framework/arguments.go (hunk 3 above)
Comment (hex108) [reply pkg/scheduler/framework/arguments.go:17]: Could use 2019 for new added files

Comment (lmzqwer2) [reply pkg/scheduler/framework/arguments.go:17]: New added files (arguments.go arguments_test.go) is now in license of 2019.

**Pull request merged**